netbox: deduplicate drf-nested-routers django4 patches

yu-re-ka opened the pull request:
they are already applied globally in our drf-nested-routers package

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

RaitoBezarius commented on pkgs/servers/web-apps/peering-manager/default.nix:
```diff
@@ -44,7 +20,7 @@ in py.pkgs.buildPythonApplication rec {
 
   format = "other";
 
-  propagatedBuildInputs = with py.pkgs; [
```
The `py → python` change is unrelated to that commit, can we separate it perhaps?

RaitoBezarius reviewed (approved):
LGTM modulo unrelated change.

RaitoBezarius commented on pkgs/servers/web-apps/peering-manager/default.nix:
```diff
@@ -7,31 +7,7 @@
 }:
 
-let
-  py = python3.override {
```
Also `peering-manager` is not `netbox`, this is confusing to have it in the same commit.

yu-re-ka commented:
Would you prefer I just leave the let-in with `let py = python3; in ...` ?

RaitoBezarius commented:
> Would you prefer I just leave the let-in with `let py = python3; in ...` ?

I am completely fine with the change, just in another commit to make the git log / blame clearer

yuyuyureka committed:
netbox: remove unnecessary python overrides

The drf-nested-routers patches are already applied globally in our
drf-nested-routers package. django 4 is already the default.

yuyuyureka committed:
peering-manager: remove unnecessary python overrides

The drf-nested-routers patches are already applied globally in our
drf-nested-routers package. django 4 is already the default.

RaitoBezarius reviewed (approved):
Feel free to merge when CI passes.
